src/altinn-app-frontend/src/components/base/MapComponent.tsx

```diff
@@ -0,0 +1,85 @@
+import * as React from 'react';
+
+import type { IComponentProps } from '..';
+import type { Location, MapLayer } from '@altinn/altinn-design-system';
+import { Map } from '@altinn/altinn-design-system';
+
+import '../../styles/shared.css';
+import {
+  getLanguageFromKey,
+  getParsedLanguageFromKey,
+} from 'altinn-shared/utils';
+import { makeStyles, Typography } from '@material-ui/core';
+
+export interface IMapComponentProps extends IComponentProps {
+  layers?: MapLayer[];
+  centerLocation?: Location;
+  zoom?: number;
+}
+
+export const useStyles = makeStyles((theme) => ({
+  footer: {
+    paddingTop: '12px',
+    color: theme.altinnPalette.primary.blueDark,
+  },
+}));
+
+export function MapComponent({
+  formData,
+  handleDataChange,
+  language,
+  readOnly,
+  layers,
+  centerLocation,
+  zoom,
+}: IMapComponentProps) {
+  const classes = useStyles();
+  const location = formData.simpleBinding
+    ? parseLocation(formData.simpleBinding)
+    : undefined;
+
+  const footerText = location
+    ? getParsedLanguageFromKey('map_component.selectedLocation', language, [
+        location.latitude,
+        location.longitude,
+      ])
+    : getLanguageFromKey('map_component.noSelectedLocation', language);
+
+  const handleMapClicked = (location: Location) => {
+    handleDataChange(`${location.latitude},${location.longitude}`);
+  };
```

Comment:
`location` is declared already in the upper scope, so to avoid confusion (and possibly complaints from SonarCloud) I would suggest to just destruct the values immediately:

```suggestion
  const handleMapClicked = ({ latitude, longitude }: Location) => {
    handleDataChange(`${latitude},${longitude}`);
  };
```